apps/sms/views/shared/js/navigation.js

```diff
@@ -491,28 +491,21 @@ function executeNavigationStep(stepName) {
 }
 
 /**
- * Attach a handler to run the `afterEnter` step after the page is loaded.
- * It uses either the custom event `navigation-transition-end` or the usual
- * `load` event.
- * @returns {Promise} Resolved once afterEnter is run.
+ * Returns promise that is resolved either once Navigation object is marked as
+ * ready (non split views) or when custom `navigation-transition-end` event is
+ * fired (split views).
+ * @returns {Promise.<void>}
  */
-function attachAfterEnterHandler() {
-  if (document.readyState === 'complete') {
-    return executeNavigationStep('afterEnter').catch(catchStepError);
-  }
-
-  var defer = Utils.Promise.defer();
+function waitForThePanelToBeReady() {
+  var readyPromise = readyDefer && readyDefer.promise || Promise.resolve();
 
-  function onNavigationEnd() {
-    window.removeEventListener('navigation-transition-end', onNavigationEnd);
-    window.removeEventListener('load', onNavigationEnd);
-    defer.resolve(executeNavigationStep('afterEnter'));
+  if (isUsingOldArchitecture()) {
```

Comment:
I think it's better in startNavigation because it's also useful when calling toPanel (from ActivityHandler for example). It's useful for both toPanel and init...